src/Day25/Solution.hs

```diff
@@ -0,0 +1,31 @@
+module Day25.Solution
+  ( crack,
+    findEncryptionKey,
+    part1,
+    transform,
+  )
+where
+
+part1 :: String -> String
+part1 = show . findEncryptionKey 7 1591838 . readPublicKeys
```

Comment:
TODO

I still don't know how to solve equations with `mod` in them, for example, solve `(7^x) mod 20201227 = 13135480` for `x`.

However, WolframAlpha does: 

https://www.wolframalpha.com/input/?i=%287%5Ex%29+mod+20201227%3D13135480

So, for now, I'm hardcoding the value.